### ISSUE TITLE
Set cachecontrol for frequently hit static endpoints

### ISSUE THREAD

**Issue by denis-yuen** [dockstore/dockstore#1834]
### Desired behaviour

Experiment with https://www.dropwizard.io/1.0.1/dropwizard-jersey/apidocs/io/dropwizard/jersey/caching/CacheControl.html based on kibana results to identify frequent unchanging endpoints. Test with lighthouse lab audits .

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/DOCK-416)
┆Issue Type: Story
┆Fix Versions: Dockstore 1.9.X
┆Sprint: Sprint 40 Octopus
┆Issue Number: DOCK-416

**Comment by denis-yuen** [comment]
Also consider use of caching certain results in database, memcached, or redis

**Comment by denis-yuen** [comment]
https://redislabs.com/lp/memcached-java/

**Comment by coverbeck** [comment]
Cache control doc is now here: https://www.dropwizard.io/en/latest/manual/core.html#caching

**Comment by coverbeck** [comment]
The problem with having the browser cache API calls is how do you invalidate the cache?

For example, one obvious candidate to cache is `/api/metadata/config.json`, as the UI blocks loading until that request is complete. It's really fast already for me, 40-100ms, but caching makes it even faster, down to 3ms.

The problem is going to be that one day we will release a new version of Dockstore that requires additional data in the config.json that won't be present in the cached copy.

While this will be rare, given the frequency of releases where the config.json changes, it has the potential to be a bad experience for users who do run into it, with no obvious workaround, unless the user knows how to use browser dev tools and disable cache.

Explored ways around this, but I don't think either will work:

1. Get the cached copy, then fetch it again when that is complete; gives the user a quick response, but then we update it later. The problem is this needs to be done with an [http interceptor](https://stackoverflow.com/questions/53232382/how-to-disable-caching-with-httpclient-get-in-angular-6), which starts getting tricky because how do you know in a global interceptor which whether you want the cached result or not?
2. Try and invalidate the cache. There is the [HTML 5 Cache Web API](https://developer.mozilla.org/en-US/docs/Web/API/Cache), but it only works in a Service Worker on Chrome in my tests. There is [window.reload(true)](https://stackoverflow.com/questions/5721704/window-location-reload-with-clear-cache), but that (old) answer says it doesn't work on Chrome.

Bottom line; I don't think this is worth pursuing.

1. The performance gains are fairly minimal for the few endpoints I identified.
2. The potential for having a user messed up with any obvious recourse is not worth it.

As an alternative, I suggest revisiting service workers, where the service worker can both cache certain endpoints and invalidate them itself. Ryan tried a while ago, and ran into issues over a year ago, but maybe both browser and Angular support of this has since improved.

**Comment by denis-yuen** [comment]
Seems like a reasonable assessment to me, I think we have bigger fish to fry
